substrate/frame/indices/src/lib.rs

```diff
@@ -264,20 +315,18 @@ pub mod pallet {
 					*maybe_value = Some((account, old_amount, perm));
 					return Ok(Pays::Yes.into());
 				} else if new_amount > old_amount {
-					// Need to reserve more
+					// Need to hold more
 					let extra = new_amount.saturating_sub(old_amount);
-					T::Currency::reserve(&who, extra)?;
+					T::NativeBalance::hold(&HoldReason::DepositForIndex.into(), &who, extra)?;
 				} else if new_amount < old_amount {
-					// Need to unreserve some
+					// Need to release some
 					let excess = old_amount.saturating_sub(new_amount);
-					let remaining_unreserved = T::Currency::unreserve(&who, excess);
-					// Defensive logging if we can't unreserve the full amount.
-					if !remaining_unreserved.is_zero() {
-						defensive!(
-							"Failed to unreserve full amount. (Index, Requested, Actual): ",
-							(index, excess, excess - remaining_unreserved)
-						);
-					}
+					let _ = T::NativeBalance::release(
+						&HoldReason::DepositForIndex.into(),
+						&who,
+						excess,
+						Precision::Exact,
+					);
```

Comment:
better to return an error if this fails IMO.

I think this can be triggered if the free balance is 0 and excess is less than E.D. for an account with some other provider reference.